parsec/CMakeLists.txt

```diff
@@ -236,9 +244,116 @@ if( BUILD_PARSEC )
   install(TARGETS parsec
     EXPORT parsec-targets
     ARCHIVE DESTINATION ${PARSEC_INSTALL_LIBDIR}
-    LIBRARY DESTINATION ${PARSEC_INSTALL_LIBDIR}
-    PUBLIC_HEADER DESTINATION ${PARSEC_INSTALL_INCLUDEDIR})
+    LIBRARY DESTINATION ${PARSEC_INSTALL_LIBDIR})
+    # Cmake < 3.20 does not support hierarchical PUBLIC and PRIVATE _HEADER,
+    # so we need to provide our own support by crafting a similar property
+    # PUBLIC_HEADER_H and PRIVATE_HEADER_H
+    # Extract these properties and properly install the files respecting the
+    # defined hierarchy
+    get_target_property(_public_headers parsec PUBLIC_HEADER_H)
+    foreach(_FILE ${_public_headers})
+        get_filename_component(_DIR ${_FILE} DIRECTORY)
+        INSTALL(FILES ${_FILE} DESTINATION ${PARSEC_INSTALL_INCLUDEDIR}/parsec/${_DIR})
+    endforeach()
+    get_target_property(_private_headers parsec PRIVATE_HEADER_H)
+    foreach(_FILE ${_private_headers})
+        get_filename_component(_DIR ${_FILE} DIRECTORY)
+        INSTALL(FILES ${_FILE} DESTINATION ${PARSEC_INSTALL_INCLUDEDIR}/parsec/${_DIR})
+    endforeach()
```

Comment:
The indentation is off here, that threw me off while trying to understand what it's doing